### ISSUE TITLE
Problem with quoted strings

### ISSUE THREAD

**Issue by IARI** [erocarrera/pydot#132]
Consider the following graph woith a quoted label:

```
digraph G {
    a [label="Graph with quoted dash \"-\" string"];
}
```

it should look like this:

![rendered version](http://g.gravizo.com/g?
digraph G {
    a [label="Graph with quoted dash \\"-\\" string"];
}
)

However, dot_parser doesn't recognize it, producing

> pyparsing.ParseException: Expected "}" (at char 18), (line:2, col:7)

**Comment by johnyf** [comment]
A patch would be welcome.

At a higher level of planning, I think that `dot_parser` should be replaced with a PLY-based parser, because:

1. `dot_parser` anyway needs to be rewritten (witness to that are the issues reported here), and
2. `pyparsing` is for top-down parsing, which is less efficient than bottom-up parsing. So, lex-yacc (`ply`) should replace `pyparsing` in `dot_parser` (also resulting in much cleaner code).

There shouldn't be much need to write a `ply` parser for DOT from scratch, because `grandalf` contains one (https://github.com/bdcht/grandalf/blob/master/grandalf/utils/dot.py). In the worst case, it would require some rewriting. However, `grandalf` has a different license (so borrowing code involves first asking for dual licensing).

Secondly, this poses the question of package architecture: where should the DOT parser be, in `pydot`, in `grandalf`, who imports whom then? Should we merge the two packages? I think that `grandalf`'s purpose is defined somewhat differently. `grandalf` is a pure Python implementation of graph layout algorithms, so a substitute for (a subset of) GraphViz.

It would be great to call `grandalf` from here, when GraphViz is missing. In other words, it would be reasonable to regard `grandalf` as a drawing backend. So, it would be `pydot` importing `grandalf`, not the other way around. In principle they could also be merged, but the usefulness of `pydot` as it is now I think is related more to its long history, so wide use, thus the API it defines. A completely restructured API (e.g., using `networkx`) would be preferable, but that would be equivalent to writing a new package, rather than changing `pydot`. So, we could identify `pydot` with (most of) its API, and let fresh architecture be developed in other packages.
For now, I think that it would be better to focus on the parser issues, and let a pure Python drawing backend as future consideration.

Taking this argument into consideration, I think that it makes more sense to ask `grandalf` to *extract* their `ply`-based DOT parser as a separate package, and share it between the two packages. This would make the parser available also to others, independently of drawing algorithms. We could then add that parser package as a requirement here, and replace `dot_parser`.

**Comment by johnyf** [comment]
Addressed by #132, see comments there.
